server/elasticsearch/src/main/java/org/hawkular/apm/server/elasticsearch/SpanServiceElasticsearch.java

```diff
@@ -0,0 +1,212 @@
+
+            client.getClient()
+                    .admin()
+                    .indices()
+                    .refresh(refreshRequestBuilder.request())
+                    .actionGet();
+
+            QueryBuilder query = QueryBuilders.termQuery("parentId", id);
+
+            SearchRequestBuilder request = client.getClient()
+                    .prepareSearch(index)
+                    .setTypes(SPAN_TYPE)
+                    .setSearchType(SearchType.DFS_QUERY_THEN_FETCH)
+                    .setQuery(query);
+
+            SearchResponse response = request.execute()
+                    .actionGet();
+
+            for (SearchHit searchHitFields : response.getHits()) {
+                try {
+                    spans.add(deserialize(searchHitFields.getSourceAsString(), Span.class));
+                } catch (IOException ex) {
+                    log.errorFailedToParse(ex);
+                    throw new IllegalStateException("Failed to deserialize span", ex);
+                }
+            }
+        } catch (IndexMissingException ex) {
+            log.errorf("No index[%s] found, so unable to retrieve spans", index);
+        }
+
+        log.tracef("Get children with parentId[%s] is: %s", id, spans);
+        return spans;
+    }
+
+    @Override
+    public void storeSpan(String tenantId, List<Span> spans) throws StoreException {
+        storeSpan(tenantId, spans, span -> span.getId());
+    }
+
+    @Override
+    public void storeSpan(String tenantId, List<Span> spans, Function<Span, String> spanIdSupplier)
+            throws StoreException {
+
+        client.initTenant(tenantId);
+
+        BulkRequestBuilder bulkRequestBuilder = client.getClient().prepareBulk();
+
+        for (Span span : spans) {
+            String json;
+            try {
+                json = serialize(span);
+            } catch(IOException ex){
+                log.errorf("Failed to serialize span %s", span);
+                throw new StoreException(ex);
+            }
+
+            log.tracef("Storing span: %s", json);
+            // modified id is used in index
+            final String modifiedId = spanIdSupplier.apply(span);
+
+            bulkRequestBuilder.add(client.getClient()
+                    .prepareIndex(client.getIndex(tenantId), SPAN_TYPE, modifiedId)
+                    .setSource(json));
+        }
+
+        BulkResponse bulkItemResponses = bulkRequestBuilder.execute().actionGet();
+
+        if (bulkItemResponses.hasFailures()) {
+            log.tracef("Failed to store traces to elasticsearch: %s", bulkItemResponses.buildFailureMessage());
+            throw new StoreException(bulkItemResponses.buildFailureMessage());
+        }
+
+        log.trace("Success storing traces to elasticsearch");
+    }
+
+    @Override
+    public void clear(String tenantId) {
+        String index = client.getIndex(tenantId);
+
+        boolean indexExists = client.getClient()
+                .admin()
+                .indices()
```

Comment:
Couldn't these three lines be stored in a local var? I can see this code changing in the future, and only one of the lines being updated :)